mobile_marketplace/bids/models.py

```diff
@@ -0,0 +1,31 @@
+from django.db import models
+from mobile_marketplace.mobiles.models import Mobile
+from mobile_marketplace.users.models import User
+from django_extensions.db.models import TimeStampedModel
+from mobile_marketplace.bids.choices import BidStateTypes
+from django_fsm import FSMField, transition
+from mobile_marketplace.bids.utils import has_valid_bid
+
+
+class Bid(TimeStampedModel):
+    amount = models.DecimalField(max_digits=10, decimal_places=2)
+    state = FSMField(default=BidStateTypes.PENDING, choices=BidStateTypes)
+
+    mobile = models.ForeignKey(Mobile, on_delete=models.CASCADE, related_name="bids")
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="bids")
+
+    @transition(field=state, source=BidStateTypes.PENDING, target=BidStateTypes.ACCEPTED, conditions=[has_valid_bid])
+    def accept_bid(self):
+        if has_valid_bid(self):
+            self.mobile.is_sold = True
+            self.mobile.save()
```

Comment:
As discussed earlier always use `edit_fields` with save to save changes for specific field.